components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/systemNotifications/AbstractWFNotifier.java

```diff
@@ -0,0 +1,205 @@
+     * @return a NotificationEndUserDTO object
+     */
+    protected NotificationEndUserDTO getPrimaryUser(String primaryUser, String organization, String portalToDisplay) {
+        NotificationEndUserDTO endUser = new NotificationEndUserDTO();
+        if (primaryUser != null) {
+            endUser.setDestinationUser(primaryUser);
+            endUser.setOrganization(organization);
+            endUser.setPortalToDisplay(portalToDisplay);
+        }
+        return endUser;
+    }
+
+    /**
+     * This method will return the UUID of a given API
+     *
+     * @param apiName    Name of the API
+     * @param apiContext Context of the API
+     * @param apiVersion Version of the API
+     * @return the UUID of the API
+     */
+    protected String getAPIUUIDUsingNameContextVersion(String apiName, String apiContext, String apiVersion,
+            String organization) throws APIManagementException {
+        return NotificationDAO.getInstance()
+                .getAPIUUIDUsingNameContextVersion(apiName, apiContext, apiVersion, organization);
+    }
+
+    /**
+     * This method will return the list of subscribers of a given API
+     *
+     * @param apiUUID            UUID of the API
+     * @param organization       Organization of the API
+     * @param apiIdEmailReplaced APIIdentifier object with email replaced provider name
+     * @return the subscribers of the API
+     */
+    protected List<SubscribedAPI> getSubscribersOfAPI(String apiUUID, String organization,
+            APIIdentifier apiIdEmailReplaced) throws APIManagementException {
+        List<SubscribedAPI> subscribedAPIs = new ArrayList<>();
+        Set<String> uniqueUserSet = new HashSet<>();
+        try {
+            UserApplicationAPIUsage[] allApiResult = ApiMgtDAO.getInstance()
+                    .getAllAPIUsageByProviderAndApiId(apiUUID, organization);
+            for (UserApplicationAPIUsage usage : allApiResult) {
+                for (SubscribedAPI apiSubscription : usage.getApiSubscriptions()) {
+                    APIIdentifier subsApiId = apiSubscription.getAPIIdentifier();
+                    APIIdentifier subsApiIdEmailReplaced = new APIIdentifier(
+                            APIUtil.replaceEmailDomain(subsApiId.getProviderName()), subsApiId.getApiName(),
+                            subsApiId.getVersion());
+                    if (subsApiIdEmailReplaced.equals(apiIdEmailReplaced)) {
+                        String userIdentifier = apiSubscription.getSubscriber()
+                                .getName() + "-" + apiSubscription.getOrganization();
+                        if (uniqueUserSet.add(userIdentifier)) {
+                            subscribedAPIs.add(apiSubscription);
+                        }
+                    }
+                }
+            }
+        } catch (APIManagementException e) {
+            APIUtil.handleException("Error while getting API usage by API ID - getAPIUsageByAPIId()", e);
```

Comment:
Add API ID as the context information for the message. Check all the messages and add context information.